orangecontrib/geo/widgets/owmap.py

```diff
@@ -167,9 +168,14 @@ def _add_controls(self):
 
     def check_data(self):
         super().check_data()
+        self.Warning.no_lat_lon_vars.clear()
+        self._attr_lat, self._attr_lon = None, None
         if self.data is not None:
-            if not self.data.domain.has_continuous_attributes(True, True):
-                self.Warning.no_continuous_vars()
+            self._attr_lat, self._attr_lon = \
+                find_lat_lon(self.data, filter_hidden=True)
+            if self._attr_lat is None or self._attr_lon is None:
+                # we either find both or none
+                self.Warning.no_lat_lon_vars()
```

Comment:
Same code, same comment as in #100. When setting `self.data` to `None`, I think the widget should also clean everything else, as a matter of hygiene. Hence, I think you should set `attr_lan` and `attr_lon` to `None`.

Also, consider upgrading `no_lat_lon_vars` from a warning to an error.